### PR TITLE
Use openshift base domain if ingress hostname is missing

### PR DESCRIPTION
**Pull request by pavolloffay**
Docker image: docker.io/pavolloffay/opentelemetry-operator:dev-756ef72-1693233303 

Updates: https://github.com/open-telemetry/opentelemetry-operator/issues/1967 
Updates: https://github.com/open-telemetry/opentelemetry-operator/issues/1969

### CODE REVIEW

**Comment by pavolloffay** [comment]
@frzifus @jaronoff97 could you please review?

File: internal/manifests/collector/collector.go [hunk 1]
```diff
@@ -60,7 +63,18 @@ func Build(params manifests.Params) ([]client.Object, error) {
 			resourceManifests = append(resourceManifests, res)
 		}
 	}
-	routes := Routes(params.Config, params.Log, params.Instance)
+
+	baseDomain := ""
+	if params.Instance.Spec.Ingress.Hostname == "" {
+		domain, err := openshift.GetOpenShiftBaseDomain(context.Background(), params.Client)
+		if err != nil {
+			params.Recorder.Event(&params.Instance, "Error", "Normal", "Failed to get basedomain for Route.")
```
Comment (frzifus) [inline internal/manifests/collector/collector.go:71]: Does it make sense to return also an error in that case?

File: controllers/opentelemetrycollector_controller_test.go [hunk 2]
```diff
@@ -60,10 +61,11 @@ func TestNewObjectsOnReconciliation(t *testing.T) {
...
 	reconciler := controllers.NewReconciler(controllers.Params{
-		Client: k8sClient,
-		Log:    logger,
-		Scheme: testScheme,
-		Config: cfg,
+		Client:   k8sClient,
+		Log:      logger,
+		Scheme:   testScheme,
+		Config:   cfg,
+		Recorder: record.NewFakeRecorder(10),
```
[the remark this answers is not in the document]
Comment (frzifus) [reply controllers/opentelemetrycollector_controller_test.go:68]: I see, thanks - was just out of interest.

File: internal/manifests/collector/collector.go (hunk 1 above)
Comment (frzifus) [reply internal/manifests/collector/collector.go:71]: hm.. I assume failing early is good. May I ask, whats your thoughts on that @jaronoff97 ?

**Pull request closed without merging**